packages/editor/src/store/selectors.js

```diff
@@ -196,6 +196,29 @@ export function getPostEdits( state ) {
 	return state.editor.present.edits;
 }
 
+/**
+ * Returns a new reference when edited values have changed. This is useful in
+ * inferring where an edit has been made between states by comparison of the
+ * return values using strict equality.
+ *
+ * @example
+ *
+ * ```
+ * const hasEditOccurred = (
+ *    getReferenceByDistinctEdits( beforeState ) !==
+ *    getReferenceByDistinctEdits( afterState )
+ * );
+ * ```
+ *
+ * @param {Object} state Editor state.
+ *
+ * @return {*} A value whose reference will change only when an edit occurs.
+ */
+export const getReferenceByDistinctEdits = createSelector(
+	() => [],
```

Comment:
Nice trick :) 

This PR assumes we track `isDirty` by the presence of `edits`, right? Can it create false positives if it's not the case?